test/framework/src/main/java/org/opensearch/test/rest/yaml/section/MatchAssertion.java

```diff
@@ -110,6 +111,25 @@ protected void doAssert(Object actualValue, Object expectedValue) {
         }
 
         if (expectedValue.equals(actualValue) == false) {
+            if (expectedValue instanceof ArrayList
```

Comment:
The sort part of response may be parsed as BigInteger, but the expect value in yaml is parsed as int/long.